src/nimlsp.nim

```diff
@@ -635,4 +661,13 @@ proc main(){.async.} =
       warnLog "Got exception: ", e.msg
       continue
 
-waitFor main()
+when defined(windows):
```

Comment:
Is there a particular reason why you've chosen to pass these in to main and therefore pass them around everywhere? You could just as easily have defined `ins` and `outs` as global variables like this as before, and then have the `multisyncTask` check the global variables without having to pass the extra argument.

Reply:
multisync may complain about that, I have face some error, multisync not work properly, and it's better for unit test functions.

Reply:
Multisync shouldn't complain, but I get the point about unit testing. I just don't like the repetition in `multisyncTask outs: outs.something`. But I guess without having them as global the only option would be a macro that inserted the left hand side as the first argument so you could at least write `outs.multisyncTask something()`

Reply:
yeah, I have to take time recall my macro knowledge , now dont need extra outs pass in macro `multisyncTask: outs.sendJson resp`